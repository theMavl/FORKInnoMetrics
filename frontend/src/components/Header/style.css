.header {
    min-height: var(--header-height);
    padding: 10px var(--page-padding);
    display: flex;
    align-items: baseline;
    justify-content: flex-start;
    flex-wrap: wrap;
    color: rgba(255, 255, 255, 1);
    background-color: rgba(var(--header-color), 1);
}

@media all and (max-width: 450px) {
    .header {
        justify-content: center;
    }
    .header :nth-child(1) {
        order: 1;
    }

    .header :nth-child(2) {
        order: 3;
    }

    .header :nth-child(3) {
        order: 2;
    }
}

.headerItem > a {
    color: rgba(255, 255, 255, 1);
    text-decoration: none;
    cursor: pointer;
}

.logoItem {
    composes: headerItem;
    padding-right: 10px;
}