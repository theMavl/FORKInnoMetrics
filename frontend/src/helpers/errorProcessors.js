import _ from 'lodash'
import {store} from '../store'
import { TYPES as USER_TYPES } from '../store/user/actionTypes'

export const processApiError = (error) => {
console.log(error.response)
  switch(_.get(error.response, 'status', 0)){
    case 0:
      return {status: 0, statusText: 'No Internet connection', data: { message: 'No Internet connection'}}
    case 401:
      store.dispatch({type: USER_TYPES.UNAUTHORIZED})
      return error.response
    default:
      return error.response
  }
}

export const getResponseError = (status, path, method) => {
  if(status === 0)
    return 'No response'
  if(status === 500)
    return 'Some error occurred'

  if(path === '/login' && method === 'post') {
    switch (status) {
      case 200:
        return 'Logged in successfully'
      case 400:
        return 'Some error occurred'
      case 401:
        return 'Wrong email or password'
      case 404:
        return 'Wrong email or password'
    }
  }
  if(path === '/activity' && method === 'post') {
    switch (status) {
      case 201:
        return 'Activity added successfully'
      case 400:
        return 'Wrong parameters provided'
    }
  }
  if(path === '/activity' && method === 'delete') {
    switch (status) {
      case 200:
        return 'Activity deleted successfully'
      case 400:
        return 'Some error occurred'
      case 404:
        return 'No such activity'
    }
  }
  if(path === '/activity' && method === 'get') {
    switch (status) {
      case 200:
        return 'Got activities successfully'
      case 404:
        return 'No activities yet'
    }
  }
  if(path === '/logout' && method === 'post') {
    switch (status) {
      case 200:
        return 'Logged out successfully'
    }
  }
  if(path === '/user' && method === 'delete') {
    switch (status) {
      case 200:
        return 'User deleted successfully'
    }
  }
  if(path === '/user' && method === 'post') {
    switch (status) {
      case 200:
        return 'User registered successfully'
      case 400:
        return 'Wrong parameters'
      case 409:
        return 'User with this email already exists'
    }
  }

  return 'Unknown error'
}