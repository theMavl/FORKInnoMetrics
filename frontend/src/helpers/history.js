import {createBrowserHistory as createHistory } from 'history'


/* FRONTEND_ADDRESS initialized by process.env in 'webpack.config.js' */
export const history = createHistory({ basename: FRONTEND_ADDRESS })