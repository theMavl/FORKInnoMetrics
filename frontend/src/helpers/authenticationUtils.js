import { history } from './history'
import _ from 'lodash'

export const isTokenExists = () => {
  return null != localStorage.getItem('user')
      && null != JSON.parse(localStorage.getItem('user')).token
}

export const getToken = () => {
  if(isTokenExists()){
    return JSON.parse(localStorage.getItem('user')).token
  }
  return ''
}

export const saveUserToLocalStorage = (token) => {
  localStorage.setItem('user', JSON.stringify({token: token}))
}

export const removeUserFromLocalStorage = () => {
  localStorage.removeItem('user')
}

export const redirectFromAuth = () => {
  let nextLocation = _.get(history, 'location.state.from', '/dashboard')
  if(nextLocation === '/login' || nextLocation === '/register')
    nextLocation = '/dashboard'
  history.push(nextLocation)
}