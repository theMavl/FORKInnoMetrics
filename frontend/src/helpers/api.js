import axios from 'axios'
import { getToken } from './authenticationUtils'
import { processApiError } from './errorProcessors'

/* DOMAIN_ADDRESS initialized by process.env in 'webpack.config.js' */

const request = axios.create({
  baseURL: DOMAIN_ADDRESS,
  headers: {
    'Content-Type': 'multipart/form-data'
  }
})

export function postRequest (url, params={}, withToken=false) {

  const formData = new FormData()
  Object.keys(params).forEach(key => formData.append(key, params[key]))
 
  return new Promise((resolve, reject) => {
    request.post(
      url,
      formData,
      {
        headers: additionalHeaders(withToken),
        withCredentials: true
      }
    )
        .then((response) => {
          resolve(response)
        })
        .catch((error) => {
          reject(processApiError(error))
        })
  })
}

export function getRequest (url, params={}, withToken=false) {

  return new Promise((resolve, reject) => {
    request.get(
      url,
      {
        params: params,
        headers: additionalHeaders(withToken),
        withCredentials: true
      }
    )
        .then((response) => {
          resolve(response)
        })
        .catch((error) => {
          reject(processApiError(error))
        })
  })
}

const additionalHeaders = (isWithToken) => isWithToken ? {'X-API-Key': getToken()} : {}
