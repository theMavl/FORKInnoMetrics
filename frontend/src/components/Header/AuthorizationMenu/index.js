import React from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'
import { NavLink } from 'react-router-dom'
import {logoutRequest} from '../../../store/user/actionCreators'
import {userAuthorized} from '../../../helpers/selectors'
import styles from './style.css'

class AuthMenu  extends React.Component {
  shouldComponentUpdate(nextProps, nextState){
    return nextProps.authorized !== this.props.authorized
  }
  render(){
    return this.props.authorized ?
        (
            <div className={styles.authMenu}
                 onClick={this.props.logout}
            >
              <div className={styles.authItem}>Logout</div>
            </div>
        )
        :
        (
            <div className={styles.authMenu}>
              <div className={styles.authItem}
                   // onClick={() => this.props.history.push('/login')}
              >
                <NavLink to='/login'>Login</NavLink>
              </div>
              <span style={{padding: '0 5px 0 5px'}}> or </span>
              <div className={styles.authItem}
                   // onClick={() => this.props.history.push('/register')}
              >
                <NavLink to='/register'>Register</NavLink>
              </div>
            </div>
        )
  }
}

const AuthorizationMenu = connect(
    (state) => ({authorized: userAuthorized(state)}),
    (dispatch) => ({
      logout: () => dispatch(logoutRequest())
    })
)(AuthMenu)

export default withRouter(AuthorizationMenu)