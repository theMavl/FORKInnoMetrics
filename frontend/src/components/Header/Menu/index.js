import React from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'
import { NavLink } from 'react-router-dom'
import {userAuthorized} from '../../../helpers/selectors'
import styles from './style.css'

class HeaderMenu extends React.Component {
  guestNavigation = [
    // {name: 'Main Page', path: '/'}
  ]
  userNavigation = [
    {name: 'MyPage', path: '/dashboard'},
    // {name: 'Activities', path: '/activities'},
    // {name: 'Projects', path: '/projects'},
    // {name: 'Settings', path: '/settings'}
  ]
  render() {
    const navigation = this.props.authorized ? this.userNavigation : this.guestNavigation
    return (
        <nav className={styles.menu}>
          {navigation.map((i) => {
            return <div className={styles.menuItem} key={i.name}
                    >
              <NavLink to={i.path}>{i.name}</NavLink>
                    </div>
          })}
        </nav>

    )
  }
}

const Menu = connect(
  (state) => ({authorized: userAuthorized(state)}),
  () => ({})
)(HeaderMenu)

export default withRouter(Menu)