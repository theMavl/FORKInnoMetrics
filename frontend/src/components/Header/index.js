import React from 'react'
import {withRouter} from 'react-router'
import Logo from '../Logo'
import Menu from './Menu'
import AuthorizationMenu from './AuthorizationMenu'
import styles from './style.css'

const Header = ({history}) => {
  return (
      <header className={styles.header}>
        <div className={styles.logoItem} onClick={() => {history.push('/')}}>
          <Logo/>
        </div>
        <Menu/>
        <AuthorizationMenu/>
      </header>

  )
}

export default withRouter(Header)