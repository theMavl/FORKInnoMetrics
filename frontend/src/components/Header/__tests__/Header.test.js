import React from 'react'
import TestRenderer from 'react-test-renderer'
import Header from '../index'
import { Router } from 'react-router-dom'
import { history } from '../../../helpers/history'
import createStore from 'redux-mock-store'
import { Provider } from 'react-redux'

describe('Test Header component', () => {

  it('Test render Header component', () => {

    const mockStore = createStore([])
    const initialStore = {
      user: {
        authorized: false
      }
    }
    const store = mockStore(initialStore)

    const instance = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Header/>
        </Router>
      </Provider>
    )
    expect(instance.toJSON()).toMatchSnapshot()
  })
})