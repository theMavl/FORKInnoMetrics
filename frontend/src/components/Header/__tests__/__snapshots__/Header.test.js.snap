// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component Test render Header component 1`] = `
<header>
  <div
    onClick={[Function]}
  >
    <div>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          <span>
            Inno
          </span>
          <span>
            Metrics
          </span>
        </span>
      </a>
    </div>
  </div>
  <nav />
  <div>
    <div>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
    <span
      style={
        Object {
          "padding": "0 5px 0 5px",
        }
      }
    >
       or 
    </span>
    <div>
      <a
        aria-current={null}
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
</header>
`;
